app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -80,7 +80,7 @@ def select
                   f.cpu >= min_cpu &&
                   f.hdd >= min_hdd
                 end
-              ) {|f| f.hourly_cost}
+              ) {|f| f.get_hourly_cost_for(tmpl.appliance_type.os_family)}
```

Comment:
Maybe `get_hourly_cost_for` can take `tmpl` as an argument and do `tmpl.appliance_type.os_family` internally?